pages/api/roadmap.ts

```diff
@@ -1,102 +1,102 @@
-      issues.forEach(async (issueData) => {
-        const childrenParsed = getChildren(issueData.body_html);
-        resultArray.push({
-          ...issueData,
-          labels: issueData.labels ?? [],
-          children: (childrenParsed.length > 0 && (await resolveChildren(childrenParsed))) || childrenParsed,
-        });
-        count += 1;
-        if (count === children.length) {
-          resolve(resultArray);
-        }
-      });
-    }).then((data) => {
-      return data;
-    }).catch((e) => {
-      return [];
-    })
-  });
+async function resolveChildrenWithDepth(children: ParserGetChildrenResponse[]): Promise<GithubIssueDataWithGroupAndChildren[]> {
+  try {
+    const issues = await resolveChildren(children);
+    return await Promise.all(issues.map(async (issueData): Promise<GithubIssueDataWithGroupAndChildren> => {
+      const childrenParsed = getChildren(issueData.body_html);
+      return {
+        ...issueData,
+        labels: issueData.labels ?? [],
+        children: (childrenParsed.length > 0 && (await resolveChildren(childrenParsed))) || childrenParsed,
+      }
+    }));
+  } catch (err) {
+    console.error('error:', err);
+    return [];
+  }
 };
 
-const addCompletionRate = ({ children, state }) => {
+function calculateCompletionRate ({ children, state }): number {
   if (state === 'closed') return 100;
-  if (!_.isArray(children)) return 0;
+  if (!Array.isArray(children)) return 0;
   const issueStats = Object.create({});
   issueStats.total = children.length;
-  issueStats.open = children.filter((v) => v.state == 'open').length;
-  issueStats.closed = children.filter((v) => v.state == 'closed').length;
+  issueStats.open = children.filter(({state}) => state === 'open').length;
+  issueStats.closed = children.filter(({state}) => state === 'closed').length;
   issueStats.completionRate = Number(Number(issueStats.closed / issueStats.total) * 100 || 0).toFixed(2);
 
   return issueStats.completionRate;
 };
 
-const addToChildren = (data, parent) => {
-  if (_.isArray(data) && data.length > 0) {
-    return data.map((item) => {
-      return {
-        labels: item.labels ?? [],
-        completion_rate: addCompletionRate(item),
-        due_date: getConfig(item).eta,
-        html_url: item.html_url,
-        group: item.group,
-        title: item.title,
-        state: item.state,
-        node_id: item.node_id,
-        body: item.body,
-        body_html: item.body_html,
-        body_text: item.body_text,
-        parent: { html_url: parent.html_url, title: parent.title },
-        children: addToChildren(item.children, item),
-      };
-    });
+function addToChildren(
+  data: GithubIssueDataWithGroupAndChildren[] | ParserGetChildrenResponse[],
+  parent: IssueData
+): IssueData[] {
+  if (Array.isArray(data) && data.length > 0) {
+    return data.map((item): IssueData => ({
+      labels: item.labels ?? [],
+      completion_rate: calculateCompletionRate(item),
+      due_date: getConfig(item).eta,
```

Comment:
we still need to rename this `getConfig` function tbh.. returning { eta} is pointless and we should just call it `getDueDate`